src/content/guides/tree-shaking.md

```diff
@@ -132,57 +156,21 @@ module.exports = {
     path: path.resolve(__dirname, 'dist')
 - }
 + },
-+ plugins: [
-+   new UglifyJSPlugin()
-+ ]
++ mode: "production"
 };
 ```
 
-T> Note that the `--optimize-minimize` flag can be used to insert the `UglifyJsPlugin` as well.
-
 With that squared away, we can run another `npm run build` and see if anything has changed.
 
 Notice anything different about `dist/bundle.js`? Clearly the whole bundle is now minified and mangled, but, if you look carefully, you won't see the `square` function included but will see a mangled version of the `cube` function (`function r(e){return e*e*e}n.a=r`). With minification and tree shaking our bundle is now a few bytes smaller! While that may not seem like much in this contrived example, tree shaking can yield a significant decrease in bundle size when working on larger applications with complex dependency trees.
 
 
-## Caveats
-
-Please note that webpack doesn't perform tree-shaking by itself. It relies on third party tools like [UglifyJS](/plugins/uglifyjs-webpack-plugin/) to perform actual dead code elimination. There are situations where tree-shaking may not be effective. For example, consider the following modules:
-
-__transforms.js__
-
-``` js
-import * as mylib from 'mylib';
-
-export const someVar = mylib.transform({
-  // ...
-});
-
-export const someOtherVar = mylib.transform({
-  // ...
-});
-```
-
-__index.js__
-
-``` js
-import { someVar } from './transforms.js';
-
-// Use `someVar`...
-```
-
-In the code above webpack cannot determine whether or not the call to `mylib.transform` triggers any side-effects. As a result, it errs on the safe side and leaves `someOtherVar` in the bundled code.
-
-In general, when a tool cannot guarantee that a particular code path doesn't lead to side-effects, this code may remain in the generated bundle even if you are sure it shouldn't. Common situations include invoking a function from a third-party module that webpack and/or the minifier cannot inspect, re-exporting functions imported from third-party modules, etc.
-
-The code used in this guide assumes you perform tree-shaking using UglifyJS plugin. However, there are other tools such as [webpack-rollup-loader](https://github.com/erikdesjardins/webpack-rollup-loader) that may produce different results depending on your setup.
-
-
 ## Conclusion
 
 So, what we've learned is that in order to take advantage of _tree shaking_, you must...
 
 - Use ES2015 module syntax (i.e. `import` and `export`).
+- Add a "sideEffects" entry to your project's `package.json` file.
```

Comment:
This would be _if_ the code is not affecting outside scope, not _always_ so we should add it to here.

Reply:
Actually no, you should always add a sideEffects entry or treeshaking won't work.

Reply:
Well, but if your librari does produce side effects you shouldn't, right? I don't want to give that message to package owners.

Reply:
Perhaps, but in that case set it to `true`, or provide an array with the specific files that include side effects as described in the guide.

It's better to explicitly have the sideEffects known to webpack than to try and infer it.

Reply:
It is required?

Reply:
Maybe is redundant since you already define what `sideEffects` means. I'm OK with it. Sorry for the noise.